lib/loader.ts

```diff
@@ -1,18 +1,23 @@
 import type { Loader } from "cosmiconfig";
 import { register, RegisterOptions } from "ts-node";
 
-import { TypeScriptCompileError } from "./typescript-compile-error";
+import { TypeScriptCompileError } from "./typescript-compile-error.js";
 
 export function TypeScriptLoader(options?: RegisterOptions): Loader {
   const tsNodeInstance = register({
     ...options,
     compilerOptions: { module: "commonjs" },
```

Comment:
FYI as of version `8.2.0`, Cosmiconfig no longer requires transpiled configuration to be CJS; [they now accept pure ESM configuration files when using the asynchronous API](https://github.com/cosmiconfig/cosmiconfig/blob/main/CHANGELOG.md#820).